core/health/shell/src/main/java/org/opennms/core/health/shell/HealthCheckCommand.java

```diff
@@ -59,6 +60,9 @@
     @Option(name = "-t", description = "Maximum number of milliseconds to wait before failing when waiting for a check to complete (e.g. try to establish a JMS session.")
     public long timeout = 5L * 1000L;
 
+    @Option(name = "--maxAge", description = "Maximum age of cached health check responses in milliseconds. Defaults to zero, i.e. cached responses of health checks are not considered.")
+    public long maxAgeMs = 0;
+
```

Comment:
Can we add an option or argument for `tag` here for the karaf command ?

Reply:
I also thought of that and already have something prepared. I had in mind to replace `HealthCheckConstants` that now defines the string values of tags by an `enum HealthCheckTag`. The `enum` can be used to support the completer. However, this would mean that all health checks have to be changed slightly because the `getTags` method now would return a list of `HealthCheckTag` instead of a `List<String>`. Having our recent discussion about PR size in mind, I was not sure if I could/should include that work here.

WDYT? The additional changes are straigthforward and rather mechanicual. I would be happy to include these changes here.

Reply:
We can keep List<String>, let the user provide Option.

Reply:
Can you clarify? Should we keep `List<String>`? In that case the possible tags would not be represented by an `enum` and the set of values would have to be repeated in the completer. This is what I wanted to avoid.

Reply:
Enum would be more useful if it is one `String` that we represent.  Since we can have multiple tags, it would be better to use List<String>.  I'm okay with Karaf command update in a separate PR.

Reply:
https://issues.opennms.org/browse/NMS-13592